text/0000-signals-in-lwc-templates.md

```diff
@@ -0,0 +1,196 @@
+
+### Object Reactivity
+
+Objects can be wrapped in Signals to make them reactive. However, changes to nested properties must be handled by replacing the entire object:
+
+```javascript
+import { signal } from 'signals';
+
+export default class ExampleComponent extends LightningElement {
+    position = signal({ x: 0, y: 0 });
+
+    move(x, y) {
+        this.position.value = { ...this.position.value, x, y };
+    }
+}
+```
+
+### Connecting to External Data Stores
+
+Components can integrate with external stores by using a pattern similar to the current `@wire`` mechanism:
+
+```javascript
+import { record } from 'app/stores';
+
+export default class ExampleComponent extends LightningElement {
+    currentRecord = record;
+}
+```
+
+```html
+<template>
+    <p>{currentRecord.value.Name}</p>
+</template>
+```
+
+__Note:__ if the currentRecord's value is not available yet, it might require some additional logic to handle that case. This is similar to what developers have to do today when using `@wire`._
+
+### LWC Internals
+
+LWC will NOT provide a built-in store implementation. Instead, it relies on a well defined protocol to interpret data in templates. This will allow developers to bring their own stores and integrate them with LWC components. It is very likely that LDS implements its own flavor of signals to work with LWC but the purpose of this RFC is to define the protocol, not the implementation.
+
+## Backward Compatibility
+
+There are two main concerns with backward compatibility:
+
+1. A component receiving an object from a parent component that contains a `value` getter and a `subscribe` method. If the object is considered a POJO by our current test, then LWC would incorrectly avoid wrapping the object with a read-only reactive process. This is a minor concern, but it is something to consider.
+2. A template using `x.value` as part of an interpolation where the value of `x` is not a signal, but contains a `value` getter and a `subscribe` method. In this case, the compiled code would attempt to call `subscribe`, if it fails, we can swallow the error and show a warning. In terms of functionality, everything remains the same, the `.value` is used to render the value into the DOM, and if it is reactive it will be updated automatically by triggering the re-rendering, which would call the `.value` getter again. Technically, this is not a non-backward compabitibility issue since we can swallow the error. But it is something to consider.
```

Comment:
Couldn't you be sure that it is a signal at build time using ts? This would help the engine to do less checks at runtime

Reply:
@AllanOricil No, the compiler compiles one file at a time in isolation.